clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -275,15 +275,20 @@ func (t *FileTarget) startTailing(ps []string) {
 		if _, ok := t.tails[p]; ok {
 			continue
 		}
+
 		fi, err := os.Stat(p)
 		if err != nil {
 			level.Error(t.logger).Log("msg", "failed to tail file, stat failed", "error", err, "filename", p)
+			t.metrics.totalBytes.DeleteLabelValues(p)
```

Comment:
wondering why just `totalBytes` and why not clear all metrics just like tailer's `t.cleanupMetrics()`? And good to make it as part of `defer()`? 

https://github.com/grafana/loki/blob/0728237530f9c6d11211a6f30fef2863243406d2/clients/pkg/promtail/targets/file/tailer.go#L221

Reply:
I like the idea of managing metrics together and as close to where they're used as possible. This situation is a little awkward because `totalBytes` is the only metric managed in two places: here and in the tailer. 

I'm not sure `defer` would be a good fit because the only metric this type cares about is `totalBytes` and only in the error case. To successfully cleanup, we'd have to capture the error in the defer, too. I think it could be pretty confusing.

Reply:
thanks for clarifying. I'm convinced :)